app/src/main/java/mozilla/lockbox/view/EditItemFragment.kt

```diff
@@ -49,6 +58,130 @@ class EditItemFragment : BackableFragment(), EditItemDetailView {
         super.onViewCreated(view, savedInstanceState)
         setupToolbar(view.toolbar)
+        setTextWatcher(view)
+        setUpKeyboardFocus(view)
+
+        val layouts = arrayOf(view.inputLayoutHostname, view.inputLayoutUsername,
+            view.inputLayoutPassword)
+
+        for(layout in layouts) {
+            layout.hintTextColor = context?.getColorStateList(R.color.hint_edit_text)
+            layout.setHintTextAppearance(R.style.HintText)
+        }
+    }
+
+    private fun setUpKeyboardFocus(view: View) {
+        view.inputHostname.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                closeKeyboard()
+            }
+        }
+
+        view.inputUsername.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                closeKeyboard()
+            }
+        }
+
+        view.inputPassword.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                closeKeyboard()
+                dispatcher.dispatch(ItemDetailAction.TogglePassword(displayed = false))
+            } else {
+                // show the password when it is focused
+                dispatcher.dispatch(ItemDetailAction.TogglePassword(displayed = true))
+            }
+        }
+    }
+
+    private fun setTextWatcher(view: View) {
+        view.inputHostname.addTextChangedListener(
+            buildTextWatcher(
+                view.inputLayoutHostname
+            )
+        )
+        view.inputUsername.addTextChangedListener(
+            buildTextWatcher(
+                view.inputLayoutUsername
+            )
+        )
+        view.inputPassword.addTextChangedListener(
+            buildTextWatcher(
+                view.inputLayoutPassword
+            )
+        )
+    }
+
+    private fun buildTextWatcher(errorLayout: TextInputLayout): TextWatcher {
+        return object : TextWatcher {
+            override fun beforeTextChanged(
+                charSequence: CharSequence,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
+            }
+
+            override fun onTextChanged(
+                charSequence: CharSequence,
+                start: Int,
+                before: Int,
+                count: Int
+            ) {
+            }
+
+            override fun afterTextChanged(editable: Editable) {
+                val inputText: String? = errorLayout.editText?.text.toString()
+
+                when (errorLayout.id) {
+                    R.id.inputLayoutHostname -> {
+                        // hostname cannot be null
+                        // has to have http:// or https://
+                        when {
+                            TextUtils.isEmpty(inputText)
+                                || (!URLUtil.isHttpUrl(inputText) and !URLUtil.isHttpsUrl(inputText))
```

Comment:
We don't have such a restriction in any of the Firefox browsers so I don't see why we should have that in Lockwise. The origin field is a URI, not a URL e.g. `ftp://example.com`, `file:///`, and `chrome://FirefoxAccounts` are all values that we currently store and Sync.